File: DesignPatternsDotNet/ObserverPattern/MyObservable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ObserverPattern
{
    public abstract class MyObservable
    {
        protected bool hasChanged = false;
        private List<MyObserver> observers = new List<MyObserver>();
        public void notifyAllOvservers()
        {
            foreach (var oserver in observers)
            {
                oserver.update();
            }
            hasChanged = false;
        }
        public void notifyAllOvservers(Object arg)
        {
            //Just another way to notify with arguments
            foreach (var oserver in observers)
            {
                oserver.update(arg);
            }
            hasChanged = false;
        }
        public MyObserver AddObserver
        {
            set{
                observers.Add(value);
            }
        }
        public MyObserver detachAObserver
        {
            set
            {
                observers.Remove(value);
            }
        }
        public int countObservers()
        {
            return observers.Count;
        }
        public abstract void setChange();
        public abstract void clearChange();
        public abstract bool hasChange();
        public MyObservable()
        {
        }
    }
}

File: DesignPatternsJava/src/com/observerPattern/MyObservable.java
package com.observerPattern;

import java.util.ArrayList;
import java.util.List;

public abstract class MyObservable {
	private List<MyObserver> observers = new ArrayList<MyObserver>();
	public void notifyAllOvservers(){
		for(int i=0; i<observers.size();i++){
			observers.get(i).update();
		}
	}
	public void notifyAllOvservers(Object arg){
	//Just another way to notify with arguments
		for(int i=0; i<observers.size();i++){
			observers.get(i).update(arg);
		}		
	}
	public void attachAObserver(MyObserver observer) {
		observers.add(observer);
	}
	public void detachAObserver(MyObserver observer) {	
		observers.remove(observer);
	}
	public int countObservers(){
		return observers.size();		
	}
	protected abstract void setChange();
	protected abstract void clearChange();
	protected abstract void hasChange();
	public MyObservable(){				
	}
}
